go/ql/lib/semmle/go/Packages.qll

```diff
@@ -18,13 +18,29 @@ class Package extends @package {
     )
   }
 
+  /**
+   * Gets the path of this package with the major version suffix (like "/v2")
+   * removed.
+   */
+  string getPathWithoutMajorVersionSuffix() {
+    result = this.getPath().regexpReplaceAll(majorVersionSuffixRegex(), "")
+  }
+
   /** Gets the scope of this package. */
   PackageScope getScope() { packages(this, _, _, result) }
 
   /** Gets a textual representation of this element. */
   string toString() { result = "package " + this.getPath() }
 }
 
+/**
+ * Gets a regex which matches major version suffixes.
+ *
+ * For example, this will match "/v2" followed by the end of a string or a "/"
+ * (but it won't include the end of the strong or the "/" in the match).
```

Comment:
```suggestion
 * Gets a regex that matches major version suffixes.
 *
 * For example, this will match "/v2" followed by the end of the string or a "/"
 * (but it won't include the end of the string or the "/" in the match).
```